### PR TITLE
GFORMS-1466 Implemented expression user.enrolments.[SERVICE].[IDENTIFIER].count/index

### PR DESCRIPTION
**Pull request by girishkamat**

### CODE REVIEW

File: app/uk/gov/hmrc/gform/graph/processor/UserCtxEvaluatorProcessor.scala [hunk 1]
```diff
@@ -32,8 +32,25 @@ object UserCtxEvaluatorProcessor extends IdentifierExtractor {
     (retrievals, userField) match {
       case (AuthenticatedRetrievals(_, enrolments, _, _, _), UserField.EnrolledIdentifier) =>
         authorizedEnrolmentValue(enrolments, authConfig)
-      case (AuthenticatedRetrievals(_, enrolments, _, _, _), UserField.Enrolment(sn, in)) =>
-        extractIdentifier(enrolments, sn, in)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/graph/processor/UserCtxEvaluatorProcessor.scala:36]: No one is using `extractIdentifier` now, so it should be removed

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Expr.scala [hunk 2]
```diff
@@ -202,11 +210,19 @@ object RosmProp {
   implicit val format: OFormat[RosmProp] = derived.oformat()
 }
 
-sealed trait UserField
+sealed trait UserField {
+  def fold[B](f: => B)(g: UserField.Enrolment => B)(h: => B): B =
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Expr.scala:214]: From caller point of view this seems as unsafe definition. There is nothing to say what `f` vs `h` cases stands for.

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Expr.scala (hunk 2 above)
Comment (girishkamat) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Expr.scala:214]: valid point, so it should be something like

```
def fold[B](f: UserField.AffinityGroup.type => B)(g: UserField.Enrolment => B)(h: UserField.Something.type => B): B =
```

Comment (girishkamat) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Expr.scala]: Done

File: app/uk/gov/hmrc/gform/graph/processor/UserCtxEvaluatorProcessor.scala (hunk 1 above)
Comment (girishkamat) [reply app/uk/gov/hmrc/gform/graph/processor/UserCtxEvaluatorProcessor.scala:36]: Removed, thanks

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**